automation/src/test/scala/org/broadinstitute/dsde/test/api/SnapshotAPISpec.scala

```diff
@@ -199,6 +199,47 @@ class SnapshotAPISpec extends AnyFreeSpecLike with Matchers with BeforeAndAfterA
       }
     }
 
+    "should delete BigQuery dataset when a workspace is deleted" taggedAs(Tags.AlphaTest, Tags.ExcludeInFiab) in {
+      implicit val patienceConfig: PatienceConfig = PatienceConfig(timeout = 20 seconds)
+
+      val owner = UserPool.userConfig.Owners.getUserCredential("hermione")
+      implicit val ownerAuthToken: AuthToken = owner.makeAuthToken(userLoginScopes ++ Seq("https://www.googleapis.com/auth/cloud-platform"))
+
+      withCleanBillingProject(owner) { projectName =>
+        withCleanUp {
+          // get the snapshot
+          val drSnapshot = listDataRepoSnapshots(1, owner)(ownerAuthToken)
+          val dataRepoSnapshotId = drSnapshot.getItems.get(0).getId
+
+          // create workspace
+          val workspaceName = s"${UUID.randomUUID().toString}-delete-bq-dataset"
+          Rawls.workspaces.create(projectName, workspaceName)
+
+          val snapshotName = "deleteWorkspaceDeleteBQDataset"
+          // add snapshot reference to the workspace. Under the covers, this creates the workspace in WSM and adds the ref
+          createSnapshotReference(projectName, workspaceName, dataRepoSnapshotId, snapshotName)(owner.makeAuthToken())
+
+          // validate the snapshot was added correctly: list snapshots in Rawls, should return 1, which we just added.
+          val listResponse = listSnapshotReferences(projectName, workspaceName)
+          val resources = Rawls.parseResponseAs[ResourceList](listResponse).getResources.asScala
+          resources should have size 1
+
+          // check that the bq dataset has been created
+          val workspaceId = resources.head.getMetadata.getWorkspaceId
+          val datasetName = "deltalayer_forworkspace_" + workspaceId.toString.replace('-', '_')
+          val datasetAfterCreation = getDataset(datasetName, projectName, ownerAuthToken)
+          datasetAfterCreation should not be empty
+
+          // delete workspace
+          Rawls.workspaces.delete(projectName, workspaceName)(owner.makeAuthToken())
+
+          // check that the bq dataset has been deleted
+          val datasetAfterDeletion = getDataset(datasetName, projectName, ownerAuthToken)
+          datasetAfterDeletion should be(None)
```

Comment:
I don't know how long it takes for google to delete the dataset once we've issued the request … might need to wrap this in an `eventually` to prevent it from being flaky. I don't know if that's worth doing now to stave off future flakiness or if we should only add `eventually` if/when we see flakiness